### ISSUE TITLE
[Bug]: Should ban auto balance channel

### ISSUE THREAD

**Issue by weiliu1031** [milvus-io/milvus#23726]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

for now, if during balance channel. here will came two shard on same channel, we can't prevent release happens in one shard, and search happens in another shard, so search will exit with error. 

we should ban auto balance channel until we can deal event which may happens in two shards.

**Comment by xiaofan-luan** [comment]
should we just release the old channel before have the new one?
That should solve the problem right?

**Comment by weiliu1031** [comment]
> should we just release the old channel before have the new one? That should solve the problem right?

if we release before sub  channel, there will be a period with no available shard

**Comment by xiaofan-luan** [comment]
> > should we just release the old channel before have the new one? That should solve the problem right?
> 
> if we release before sub channel, there will be a period with no available shard

Yep, but it has to be only one leader

**Comment by congqixia** [comment]
We could have system has two delegators at the same time: 
- Make old delegator work as sub node during the balance
- All load/release operation dispatched by new delegator shall be forwarded by the older one
- After the new delegator become workable, de-register the old delegator

**Comment by xiaofan-luan** [comment]
> We could have system has two delegators at the same time:
> 
> * Make old delegator work as sub node during the balance
> * All load/release operation dispatched by new delegator shall be forwarded by the older one
> * After the new delegator become workable, de-register the old delegator

It's might be doable, if querynode accept duplicated insert.
And the msgstream has to change to shared mode I guess

**Comment by weiliu1031** [comment]
we will ban balance channel for short term, until we can deal with two shard online for same channel in replica.

**Comment by yanliang567** [comment]
@weiliu1031 shall we keep this open?

**Comment by weiliu1031** [comment]
> balance channel should be the priority for 2.3

@yah01 is working on supporting two shard in same channel exist at same time. after that, we will re-enable balance channel

**Comment by yah01** [comment]
/assign
working on it

**Comment by yah01** [comment]
master fixed with #24849
